app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -33,18 +32,18 @@ def claim
       topic_id = params.require(:topic_id).to_i
       topic = Topic.find(topic_id)
 
-      assigned = TopicCustomField.where(
-        "topic_id = :topic_id AND name = 'assigned_to_id' AND value IS NOT NULL",
-        topic_id: topic_id
-      ).pluck(:value)
+      assigned_id = Assignment
+        .where(topic_id: topic_id)
+        .not(assigned_to_id: nil)
+        .pluck_first(:assigned_to_id)
 
-      if assigned && user_id = assigned[0]
-        extras = nil
-        if user = User.where(id: user_id).first
+      if assigned
```

Comment:
I think you meant `assigned_id` here?

```suggestion
      if assigned_id
```

Reply:
What a nice opportunity to write a spec 😆